Fixes #17405: Added plugin icon to plugin list/detail

jnovinger opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #17405
- uses new `icon` URL value from the [plugins catalog API](https://api.netbox.oss.netboxlabs.com/v1/plugins/)
- applies minimal styling to keep image display from being to large relative to nearby text
- assumes icons are black and white and inverts the image via CSS for dark mode

<!--
    Please include a summary of the proposed changes below.
-->

jnovinger committed:
Added plugin icon to plugin list/detail

jnovinger commented on netbox/project-static/styles/custom/_misc.scss:
```diff
@@ -75,3 +75,15 @@ span.color-label {
 }
+
+img.plugin-icon {
+  max-width: 1.4285em;
```
This was eye-balled. I didn't see any immediate variables to re-use here.

jnovinger commented on netbox/project-static/styles/custom/_misc.scss:
```diff
@@ -75,3 +75,15 @@ span.color-label {
+  // Assuming icon is black/white line art, invert it and tone down brightness
+  img.plugin-icon {
+    filter: invert(100%) brightness(80%);
+  }
```
Not loving this approach, unless we'll only ever accept black/white line-drawing-ish icons.

pheus replied:
Just a quick drive-by thought: you could consider adding `grayscale()` to the filter as well:
```css
filter: grayscale(100%) invert() brightness(80%);
```

Also, small note - you don't need to specify `100%` for `invert`; `invert()`, `invert(1)`, and `invert(100%)` are equivalent. (See [MDN docs](https://developer.mozilla.org/en-US/docs/Web/CSS/filter-function/invert) if helpful.)

jnovinger committed:
Update compiled static files with updated NPM deps

jnovinger committed:
Fix KeyError

jnovinger committed:
Add grayscale image filter

Hat tip to @pheus. Thanks!

I did end up leaving the filter function arguments as the Sass
processor complains when you try calling a filter function without
an argument. :/

bctiemann commented:
I think this looks fine code-wise, but the icon looks a little too big for the text:

<img width="572" alt="Screenshot 2025-04-30 at 8 45 19 AM" src="https://github.com/user-attachments/assets/50f0531b-e74d-4689-bdf8-aac54c3b4ea6" />

However I don't think that's a problem with the icon, but rather that the text in the table rows is too small / not vertically centered in the table rows throughout the UI. That's a separate issue that should be raised.
